RealmSwift/Sync.swift

```diff
@@ -311,12 +311,43 @@ public typealias Provider = RLMIdentityProvider
         }
     }
 
+    /// The implementation of @dynamicMemberLookup that allows for dynamic remote function calls with a `callable` return.
+    public subscript(dynamicMember string: String) -> FunctionCallable {
+        FunctionCallable(name: string, user: user)
+    }
+}
+
+/// Structure providing an interface to been able to return a `callable` for a MongoDB Realm function.
```

Comment:
I would reword this since it's not really saying anything. Why does it exist? What are we enabling? I would call out the syntactic sugar here similarly to line 763.